src/forward_link.py

```diff
@@ -0,0 +1,105 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+
+def zerofy_values(
```

Comment:
Function is clear, has excellent doc strings and type hints. Nice value error raised if an incorrect expression is given. 

Not a massive fan of the name of this function and the unit tests for this function are named `test_..._filter` which is a little confusing. maybe more applicable if called `masking_values` or something along those lines (sorry I cant give you a better name off the top of my head)

Reply:
Honestly this name puzzled me ,  masking_values  is looking good since is based on masking the dataframe anyway, i like it